plugin/src/main/groovy/com/novoda/buildproperties/internal/BuildPropertiesException.groovy

```diff
@@ -32,14 +32,23 @@ class BuildPropertiesException extends Exception {
         List<Throwable> newExceptions = new ArrayList<>(exceptions)
         List<String> newAdditionalMessages = new ArrayList<>(additionalMessages)
         if (throwable instanceof BuildPropertiesException) {
-            newExceptions.addAll(((BuildPropertiesException) throwable).exceptions)
-            newAdditionalMessages.addAll(((BuildPropertiesException) throwable).additionalMessages)
+            def other = (BuildPropertiesException) throwable
+            newExceptions.addAll(removeDuplicateExceptions(other.exceptions))
+            newAdditionalMessages.addAll(removeDuplicateAdditionalMessages(other.additionalMessages))
         } else {
-            newExceptions.add(throwable)
+            newExceptions.addAll(removeDuplicateExceptions([throwable]))
         }
         return new BuildPropertiesException(newExceptions, newAdditionalMessages)
     }
 
+    private List<Throwable> removeDuplicateExceptions(List<Throwable> others) {
+        others.findAll { !message.contains(it.message) }
+    }
+
+    private List<String> removeDuplicateAdditionalMessages(List<String> messages) {
+        messages.findAll { !additionalMessages.contains(it) }
```

Comment:
Isn't this just `messages - additionalMessages`?

Reply:
I have no idea 🙈 Groovy has so many things I haven't discovered yet. I will try your suggestion